models/aether_cube_sphere/model_mod.f90

```diff
@@ -0,0 +1,868 @@
+do i = 1, num_bound_points
+   bounding_state_index(i, 1) =  get_state_index(grid_face(i), grid_lat_ind(i), &
+      grid_lon_ind(i), below_index, var_id)
+   bounding_value(i, 1, :) = get_state(bounding_state_index(i, 1), state_handle)
+   bounding_state_index(i, 2) =  get_state_index(grid_face(i), grid_lat_ind(i), &
+      grid_lon_ind(i), above_index, var_id)
+   bounding_value(i, 2, :) = get_state(bounding_state_index(i, 2), state_handle)
+enddo
+
+! Do inverse distance weighted horizontal interpolation on both levels
+below_values = idw_interp(ens_size, RAD2DEG*pt_lat, RAD2DEG*pt_lon, &
+   RAD2DEG*grid_pt_lat, RAD2DEG*grid_pt_lon, bounding_value(:, 1, :), num_bound_points)
+above_values = idw_interp(ens_size, RAD2DEG*pt_lat, RAD2DEG*pt_lon, &
+   RAD2DEG*grid_pt_lat, RAD2DEG*grid_pt_lon, bounding_value(:, 2, :), num_bound_points)
+
+! Do the vertical interpolation, linear in height to get final
+call vert_interp(ens_size, below_values, above_values, fract, expected_obs)
+
+end subroutine model_interpolate
+
+!------------------------------------------------------------------
+
+! Returns the smallest increment in time that the model is capable 
+! of advancing the state in a given implementation, or the shortest
+! time you want the model to advance between assimilations.
+
+function shortest_time_between_assimilations()
+
+type(time_type) :: shortest_time_between_assimilations
+
+if(.not. module_initialized) call static_init_model
+
+shortest_time_between_assimilations = assimilation_time_step
+
+end function shortest_time_between_assimilations
+
+!------------------------------------------------------------------
+
+! Given an integer index into the state vector, returns the
+! associated location and optionally the physical quantity.
+
+subroutine get_state_meta_data(index_in, location, qty)
+
+integer(i8),         intent(in)            :: index_in
+type(location_type), intent(out)           :: location
+integer,             intent(out), optional :: qty
+
+! Local variables
+
+integer  :: lev_index, col_index, my_var_id, my_qty, dom_id
+real(r8) :: lat, lon
+integer  :: seconds, days ! for f10.7 location
+real(r8) :: longitude     ! for f10.7 location
+
+if(.not. module_initialized) call static_init_model
+
+call get_model_variable_indices(index_in, col_index, lev_index, &
+   no_third_dimension, var_id=my_var_id, dom_id=dom_id, kind_index=my_qty)
+
+! F10.7 scalar location does not have a column
+if(trim(get_variable_name(dom_id, my_var_id)) == 'SCALAR_F10.7') then
+   ! AETHER MODELERS SHOULD REFINE THIS AS NEEDED
+   ! Set the location as per TIEGCM example for now
+   ! f10_7 is most accurately located at local noon at equator.
+   ! 360.0 degrees in 86400 seconds, 43200 secs == 12:00 UTC == longitude 0.0
+   call get_time(state_time, seconds, days)
+   longitude = 360.0_r8 * real(seconds,r8) / 86400.0_r8 - 180.0_r8
+   if (longitude < 0.0_r8) longitude = longitude + 360.0_r8  
+   write(*, *) 'longitude for F10.7 is ', longitude
```

Comment:
should probably use the error handler for this message
E_ALLMSG to have all tasks print (since it will not necessarily be task 0 that has the f10.8 state element).